src/index.ts

```diff
@@ -23,13 +39,45 @@ export {
 	Resolver,
 };
 
+export interface ResolveListeners {
+	close?: Array<() => void>;
+	data?: Array<(chunk: Buffer | string) => void>;
+	end?: Array<() => void>;
+	error?: Array<(e: Error) => void>;
+	readable?: Array<() => void>;
+	'resolved-name'?: Array<(dockerfilePath: string) => void>;
+	resolver?: Array<(resolverName: string) => void>;
+}
+
+/**
+ * Reads bundle.tarStream (typically a reference to BuildTask.buildStream) and
+ * resolves it to a Docker-compatible tar stream (i.e. selects a specified
+ * Dockerfile, translates Dockerfile.template vars and the like).
+ * @param bundle Bundle including the input tar stream
+ * @param resolvers Bundle resolvers
+ * @param resolveListeners Event listeners for tar stream resolution.
+ * Note that although you can add listeners to the returned Pack stream,
+ * the stream is piped to before it is returned and events may be missed
+ * unless the listeners are installed through the resolveListeners argument.
+ * You should always add at least an 'error' handler, or uncaught errors
+ * may crash the app.
+ * @param dockerfile User-selected Dockerfile (docker-compose `dockerfile:`)
+ */
 export function resolveInput(
 	bundle: Bundle,
 	resolvers: Resolver[],
+	resolveListeners: ResolveListeners,
```

Comment:
This added argument is the reason for a major version bump.

Reply:
Yes, I thought about making it optional, it could be done. But then I considered that not providing this argument and thus potentially ignoring error events is "almost certainly a bug" and the function interface should discourage it. Adding this argument is main purpose of this PR. Note that before this PR, there was no reliable way of providing an error handler for the stream events. Making this arg optional would be like making the fix optional...